### DISCUSSION TITLE
Wireguard VPN video streaming: What is the minimum speed you need to make it work ?

### DISCUSSION THREAD

**Discussion by chogovadze** [photoprism/photoprism#1855] [Q&A] [answered]
Hello everyone,

I have been trying to use Wireguard with Photoprism to stream videos when I am traveling. Though I can get up to 400 Mbits through iPerf3 I seem to be unable to watch without constant buffering. When I am directly connected to my network through wifi I am able to access videos without any issues. I was wondering in your experience what was the minimum speed on which you were able to successfully playback videos through a VPN ? 

I was not able to isolate if this was due to issues with MTU, transcoding or simply not enough bandwidth.

**Comment by lastzero** [comment, marked as answer]
Videos are not downloaded all at once but streamed, so request / round-trip time plays a role as well — not only the one-directional download speed.

**Reply by chogovadze** [reply]
Thanks for reaching out! If I understood correctly you are referring to the fact that up/down speeds are often not symmetrical. If that is the case, I have tested at both sites, both up and down speeds are around 950 Mbits. Furthermore the iperf test was done from one device to another device through the VPN.

**Reply by srett** [reply]
That is odd. 400MBit/s sounds plenty enough, I can stream most videos just fine on my 10MBit/s connection, using an SSH tunnel.
Could you try accessing your PP instance directly, without the Wireguard tunnel, just for comparison, or are you actually testing the Wireguard config from within your Wifi?

You could also check the RTT of your Wireguard tunnel by running `ping -i 0.2 <wireguard_remote_ip>` and check for outliers or worse, packet loss. The delay shouldn't be much higher than without the tunnel.
